modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFieldValuesProvider.java

```diff
@@ -299,17 +298,34 @@ private List<InfoFieldValue<Object>> _getInfoFieldValues(
 
 		ThemeDisplay themeDisplay = _getThemeDisplay();
 
-		if (themeDisplay != null) {
-			objectEntryFieldValues.addAll(
-				_getObjectFieldsInfoFieldValues(
-					_getObjectEntry(
-						objectEntry.getExternalReferenceCode(),
-						_objectDefinition, themeDisplay),
-					_objectFieldLocalService.getObjectFields(
-						objectEntry.getObjectDefinitionId(), false),
-					themeDisplay));
+		if (themeDisplay == null) {
+			ServiceContext serviceContext =
+				ServiceContextThreadLocal.getServiceContext();
+
+			themeDisplay = new ThemeDisplay() {
+				{
+					setCompany(
+						_companyLocalService.getCompany(
+							serviceContext.getCompanyId()));
+					setLocale(
+						LocaleUtil.fromLanguageId(
+							serviceContext.getLanguageId()));
+					setSiteGroupId(serviceContext.getScopeGroupId());
+					setUser(
+						_userLocalService.getUser(serviceContext.getUserId()));
+				}
+			};
```

Comment:
Hey @seltong I believe it's not necessary because you're getting the service context in the method `_getThemeDisplay()` could you please take a look and tell me if it's necessary?

Thank you!

Reply:
Hey @MarinhoFeliphe, I'm getting the service context in the  `_getThemeDisplay()` but in the `_getInfoFieldValues()` I don't have access to this service context.

I thought that was not a good idea get the service context before the `_getThemeDisplay();` and pass it as a parameter to the `_getThemeDisplay();`, but following this approach I would have to pass the service context in all uses of the `_getThemeDisplay();`. It doesn't seem a good approach to me.

Another approach is create a `private ServiceContext _serviceContext` in the class, but it also doesn't seem a good approach to me because we will use it only two once.

So I was inclined to follow with the approach of get the service context again